share/spack/setup-env.sh

```diff
@@ -227,8 +227,8 @@ fi
 #
 # Find root directory and add bin to path.
 #
-_sp_share_dir=$(cd "$(dirname $_sp_source_file)" && pwd)
-_sp_prefix=$(cd "$(dirname $(dirname $_sp_share_dir))" && pwd)
+_sp_share_dir=$(command cd "$(dirname $_sp_source_file)" && pwd)
+_sp_prefix=$(command cd "$(dirname $(dirname $_sp_share_dir))" && pwd)
```

Comment:
I think I would prefer `builtin` over `command`. From what I can gather, `command` searches the `PATH` for the executable and returns the first one it finds, so you could still potentially end up with a redefinition. `command -p` is a little safer as it only searches the system paths. `builtin` on the other hand seems to only use the system command. From the manpage:

> Execute the specified shell builtin, passing it  arguments,  and
              return its exit status.  This is useful when defining a function
              whose name is the same as a shell builtin, retaining  the  func-
              tionality of the builtin within the function.  The cd builtin is
              commonly redefined this way.  The  return  status  is  false  if
              shell-builtin is not a shell builtin command.